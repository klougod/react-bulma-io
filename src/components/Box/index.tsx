import React, { ReactNode } from 'react'

interface BoxProps {
  className?: string
  children?: ReactNode
}

export const Box = ({ className, children }: BoxProps) => (
  <div className={`box ${className}`}>{children}</div>
)

Box.defaultProps = {
  className: ''
}

export default Box
