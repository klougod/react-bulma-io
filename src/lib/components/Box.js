import React from 'react'
import PropTypes from 'prop-types'

export const Box = ({ className, children }) => (
  <div className={`box ${className}`}>
    { children }
  </div>
)

Box.defaultProps = {
  className: ''
}

Box.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Box