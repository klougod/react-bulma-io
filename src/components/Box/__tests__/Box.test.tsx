import React from 'react'
import Enzyme from 'enzyme'

import Box from '../'

describe('Box component test', function () {
  it('should render without crash with props', function () {
    const boxMockProps = {
      className: 'boxClass'
    }
    const boxMock = Enzyme.shallow(<Box {...boxMockProps} />)
    expect(boxMock).toMatchSnapshot()
  })
})
