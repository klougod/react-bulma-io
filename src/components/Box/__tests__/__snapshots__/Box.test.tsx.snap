// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component test should render without crash with props 1`] = `
<div
  className="box boxClass"
/>
`;
